aten/src/ATen/native/cpu/CopyKernel.cpp

```diff
@@ -4,32 +4,65 @@
 #include <ATen/CPUApplyUtils.h>
 #include <ATen/Dispatch.h>
 #include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
 #include <ATen/native/Copy.h>
 
 namespace at {
 namespace native {
 namespace {
 
-// TODO: this number was copied from TH, test to see if it's the right number
-constexpr int64_t COPY_GRAIN_SIZE = 20000;
+template <typename self_T>
+void copy_kernel_cast_t_impl(Tensor& self, const Tensor& src) {
+  auto builder = TensorIterator::Builder();
```

Comment:
I think it's a bit better to construct the `TensorIterator` once in `_s_copy__cpu` and pass it as an argument (instead of passing the Tensors as arguments). The reasons for this are:

- It's a step towards centralizing error checks (reduces duplicate code and risk of missing shape or dtype checks for certain cases)
- It better matches the pattern of the other kernels and people are likely to use this as an example

Reply:
Good point

Reply:
On the second thought, right now TensorIterator construction and loop are staying together, and Copy.cpp doesn't need to know anything about implementation details of the kernel. Wich perfectly isolates implementation from dispatch and allows us to replace kernels (if necessary) at ease.